components/gitpod-protocol/java/src/main/java/io/gitpod/gitpodprotocol/api/GitpodServerLauncher.java

```diff
@@ -70,77 +59,52 @@ public GitpodServerConnection listen(
+                    httpClient.getProxyConfiguration().getProxies().add(new HttpProxy(hostName, port));
+                } else if (proxy.type().equals(Proxy.Type.SOCKS)) {
+                    httpClient.getProxyConfiguration().getProxies().add(new Socks4Proxy(hostName, port));
                 }
-            }).build(), URI.create(apiUrl)));
-            return connection;
-        }
-        if (sslContext == null) {
-            httpClient = new HttpClient();
-        } else {
-            SslContextFactory ssl = new SslContextFactory.Client();
-            ssl.setSslContext(sslContext);
-            httpClient = new HttpClient(ssl);
-        }
-        for (Proxy proxy : proxies) {
-            if (proxy.type().equals(Proxy.Type.DIRECT)) {
-                continue;
-            }
-            SocketAddress proxyAddress = proxy.address();
-            if (!(proxyAddress instanceof InetSocketAddress)) {
-                GitpodServerConnectionImpl.LOG.log(Level.WARNING, gitpodHost + ": unexpected proxy:", proxy);
-                continue;
-            }
-            String hostName = ((InetSocketAddress) proxyAddress).getHostString();
-            int port = ((InetSocketAddress) proxyAddress).getPort();
-            if (proxy.type().equals(Proxy.Type.HTTP)) {
-                httpClient.getProxyConfiguration().getProxies().add(new HttpProxy(hostName, port));
-            } else if (proxy.type().equals(Proxy.Type.SOCKS)) {
-                httpClient.getProxyConfiguration().getProxies().add(new Socks4Proxy(hostName, port));
             }
-        }
-        ClientContainer container = new ClientContainer(httpClient);
-        // allow clientContainer to own httpClient (for start/stop lifecycle)
-        container.getClient().addManaged(httpClient);
-        container.start();
 
-        GitpodServerConnectionImpl connection = new GitpodServerConnectionImpl(gitpodHost);
-        connection.whenComplete((input, exception) -> {
-            try {
-                container.stop();
-            } catch (Throwable t) {
-                GitpodServerConnectionImpl.LOG.log(Level.WARNING, gitpodHost + ": failed to stop websocket container:", t);
-            }
-        });
+            ClientContainer container = new ClientContainer(httpClient);
+            // allow clientContainer to own httpClient (for start/stop lifecycle)
+            container.getClient().addManaged(httpClient);
+            container.start();
+
+            connection.whenComplete((input, exception) -> {
+                try {
+                    container.stop();
+                } catch (Throwable t) {
+                    GitpodServerConnectionImpl.LOG.log(Level.WARNING, gitpodHost + ": failed to stop websocket container:", t);
+                }
+            });
+        }
 
-        connection.setSession(container.connectToServer(new Endpoint() {
+        connection.setSession(ContainerProvider.getWebSocketContainer().connectToServer(new Endpoint() {
```

Comment:
True! I'm updating the code now to bring the `container` back to this line.

But I'm starting to think that was the cause of the incident. Because the previous code was using `ContainerProvider.getWebSocketContainer()`.

Cold we do like the following?

- if hasProxies: `connection.setSession(container.connectToServer(new Endpoint() {`
- else: `connection.setSession(ContainerProvider.getWebSocketContainer().connectToServer(new Endpoint() {`

Or we shouldn't use `ContainerProvider.getWebSocketContainer()` for any case?

Reply:
> Or we shouldn't use ContainerProvider.getWebSocketContainer() for any case?

there should not be need for it.